docs/get-started/support-new-model.md

```diff
@@ -0,0 +1,548 @@
+# How to support a new model
+
+!!! todo
+    Rename titles and focus on supporting new models
+
+## How kernel optimize a model
+
+### Overview
+
+To optimize a model, kernl use [Torchdynamo](https://github.com/pytorch/torchdynamo) JIT compiler and provide a custom backend where we replace part of the [Torch FX](https://pytorch.org/docs/1.13/fx.html) graph with optimized kernels.
+
+The custom backend is defined in [src/kernl/model_optimization.py](https://github.com/ELS-RD/kernl/blob/74d26c69cd3b12b0097804735ed0555123926503/src/kernl/model_optimization.py#L31)
+
+``` { .py }
+def _compiler(gm: torch.fx.GraphModule, example_inputs: List[torch.Tensor]):
+    dynamo_backend_ofi(gm)
+    return cuda_graphs_wrapper(gm, example_inputs, pool=_pool)
+```
+This backend combines two steps:
+
+- First one is to apply the graph replacements
+- Second one is to use [CUDA graphs](https://pytorch.org/blog/accelerating-pytorch-with-cuda-graphs/)
+
+We won't elaborate the second step and focus on the first one that actually does the graph replacements.
+
+### Replace part of the FX graph
+
+First of all, a few word about [Torch FX](https://pytorch.org/docs/1.13/fx.html).
+Torch FX is a torch module to module transformation toolkit. It can trace an torch module (or a function) execution. All the operation are then recorded into a list of nodes that are represented as an FX Graph. From this graph, Torch FX generates python code matching the graph's semantic. Both graph and python code are accessible from the Torch FX GraphModule, which is also a torch module instance. Playing with Torch FX allows us to modify a graph but to stay at the torch module level.
```

Comment:
The explanation seems to be very detailed. May be something related to it traces computation graph + let us manipulate it? And a special word about dynamic behavior?

Reply:
Actually I've detailed a bit as I explain more how torch fx works afterwards (not in the draft version you saw)